tajo-client/src/main/java/org/apache/tajo/client/v2/QueryFuture.java

```diff
@@ -0,0 +1,125 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.tajo.client.v2;
+
+import org.apache.tajo.auth.UserRoleInfo;
+
+import java.sql.ResultSet;
+import java.util.concurrent.Future;
+
+public interface QueryFuture extends Future<ResultSet> {
+  /**
+   * Get a query id
+   *
+   * @return query id
+   */
+  String id();
+
+  String queue();
+
+  /**
+   * Get a query state
+   *
+   * @return query state
+   */
+  QueryState state();
+
+  /**
+   * Get a normalized progress (0 ~ 1.0f) of a query running
+   *
+   * @return progress
+   */
+  float progress();
+
+  /**
+   * A submitted or running query state is normal
+   *
+   * @return True if a query state is normal
+   */
+  boolean isOk();
+
+  /**
+   * Get whether the query is successfully completed or not.
+   *
+   * @return True if the query is successfully completed.
+   */
+  boolean isCompleted();
```

Comment:
For me, ```query is completed``` sounds that query execution is completed whether it is succeeded, failed or killed.
How about changing ```isSucceeded()```? IMHO, it looks more intuitive.